examples/AspNetCore/ControllerSample/Controllers/SampleController.cs

```diff
@@ -155,5 +155,18 @@ public ActionResult<Account> ExampleCustomTopic(Transaction transaction)
         {
             return Ok();
         }
+
+        /// <summary>
+        /// Method which uses <see cref="TopicMetadataAttribute" /> for binding this endpoint to a subscription and adds routingkey metadata.
+        /// </summary>
+        /// <param name="transaction"></param>
+        /// <returns></returns>
+        [Topic("pubsub", "topicmetadata")]
+        [TopicMetadata("routingKey", "keyA")]
+        [HttpPost("examplecustomtopicmetadata")]
+        public ActionResult<Account> ExampleCustomTopicMetadata(Transaction transaction)
+        {
+            return Ok();
+        }
```

Comment:
For examples, it's better to have code that accomplishes something instead of just showing the attribute. Is there anyway we can utilize it? Or is it too specific to RabbitMQ to make it a generic example?

Reply:
Currently, only rabbitmq is using this metadata If the subsequent proposal [#4694](https://github.com/dapr/dapr/issues/4694) can be adopted, we will have more examples that can be used.

Reply:
Two interfaces are mainly used to solve an event.
for example
```
        [Topic("pubsub", "metadata", new string[1] { "id1" })]
        [Topic("pubsub", "metadata.1", true)]
        [HttpPost("/multiMetadataTopicAttr")]
        [TopicMetadata("n1", "v1")]
        [TopicMetadata("id1", "n2", "v2")]
        [TopicMetadata("id1", "n2", "v3")]
        public void MultipleMetadataTopics()
        {
        }
```
When the metadata required by topica and topicb are inconsistent, it provides the ability to filter metadata and how to convert multiple values into a string when duplicate metadata occurs.
res:
`metadata` : n1=v1,n2=v2,v3
`metadata.1`: n1=v1